cesium/delete.go

```diff
@@ -257,8 +254,6 @@ func (db *DB) garbageCollect(ctx context.Context, maxGoRoutine int64) error {
 			return udb.GarbageCollect(_ctx)
 		})
 	}
-
-	db.mu.RUnlock()
 	return sCtx.Wait()
```

Comment:
I feel like we shouldn't lock down channel creation during GC. Creating a channel that uses an index channel in GC as index is fine – we just won't be able to write data for it.

As for deletion, we should not allow that, but we should address that at the Unary level with `entityCount`